src/Control/Optics/Linear/Internal.hs

```diff
@@ -123,9 +122,8 @@ build (Optical l) x = Linear.runCoKleisli (l (Linear.CoKleisli getConst')) (Cons
 getConst' :: Const a b ->. a
 getConst' (Const x) = x
 
--- TODO: think about if this is the right constraint
-lengthOf :: P.Num r => Optic_ (NonLinear.Kleisli (Const (Sum r))) a b s t -> s -> r
-lengthOf l s = getSum (gets l (const (Sum 1)) s)
+lengthOf :: MultIdentity r => Optic_ (NonLinear.Kleisli (Const (Adding r))) a b s t -> s -> r
```

Comment:
I'm confused. Shoudn't it be `AddIdentity`? How does it even type?

Reply:
Surprisingly not, the `MultIdentity` is only necessary so I can say `one`, there's no actual requirement here that `Adding r` forms a `Monoid`. But of course whenever this is used, `AddIdentity r` is also required.

Reply:
Oh. I see. Because if it's a lens, then you just need `1`. If you are looking at a traversal, then you will have the additional requirement that `NonLinear.Kleisli (Const (Sum r))` is an applicative, which, in turn, requires that `Sum r` is a monoid, and that `r` is `AddIdentity`.

Neat.

Reply:
Right, pretty much.